test_common/harness/imageHelpers.cpp

```diff
@@ -2765,10 +2765,7 @@ void pack_image_pixel(float *srcVector, const cl_image_format *imageFormat,
         case CL_SIGNED_INT32: {
             cl_int *ptr = (cl_int *)outData;
             for (unsigned int i = 0; i < channelCount; i++)
-                ptr[i] = (int)CONVERT_INT(
-                    srcVector[i], MAKE_HEX_FLOAT(-0x1.0p31f, -1, 31),
-                    MAKE_HEX_FLOAT(0x1.fffffep30f, 0x1fffffe, 30 - 23),
-                    CL_INT_MAX);
+                ptr[i] = round_to_even(srcVector[i]);
```

Comment:
Since round_to_even has clamping logic inside it, it works as intended for 32 bit types. The rest of the cases are basically hacks to correct the clamping behavior for different types. A long-term solution would be to templatize round_to_even to work correctly with all types.

Reply:
It might be good to create an issue so we don't forget to revisit this.

Reply:
Done #1395

Reply:
Thanks!